### ISSUE TITLE
"Load File in REPL" stopped working

### ISSUE THREAD

**Issue by MrGung** [cursive-ide/cursive#2927]
Using cursive 1.13.3-2024.1 and bb 1.3.190 with IntelliJ IDEA 2024.1.3 (Ultimate Edition). After getting Exceptions for some time (see https://github.com/cursive-ide/cursive/issues/2896) when trying to load a file, now nothing happens (no excpeptions but the file also does not get loaded).

Is there a log I could provide?

**Comment by MrGung** [comment]
I updated to 1.13.4-eap7-2024.1, this gives

`Error evaluating - class java.lang.IllegalArgumentException: Argument for @NotNull parameter 'filter' of com/intellij/util/indexing/FileBasedIndexEx.getValues must not be null`

**Comment by cursive-ide** [comment]
Could you attach your log please? You'll find it in the Help menu under _Help | Show Log in Finder/Explorer_.

**Comment by MrGung** [comment]
I think this is the relevent part:

```
2024-07-19 08:32:56,259 [6280048]   INFO - #cursive.repl.process-runner - Received nREPL connection details: 127.0.0.1:53214
2024-07-19 08:33:02,990 [6286779]   INFO - STDERR - org.jetbrains.concurrency.AsyncPromise$Companion$CANCELED$1
2024-07-19 08:33:03,307 [6287096]   INFO - STDERR - org.jetbrains.concurrency.AsyncPromise$Companion$CANCELED$1
2024-07-19 08:33:19,846 [6303635] SEVERE - #cursive.repl - Error evaluating
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'filter' of com/intellij/util/indexing/FileBasedIndexEx.getValues must not be null
 at com.intellij.util.indexing.FileBasedIndexEx.$$$reportNull$$$0 (FileBasedIndexEx.java:-1)
    com.intellij.util.indexing.FileBasedIndexEx.getValues (FileBasedIndexEx.java:-1)
    cursive.index$namespace_names.invokeStatic (index.clj:489)
    cursive.index$namespace_names.invoke (index.clj:483)
    cursive.namespace$dependency_map$reify__4616.compute (namespace.clj:42)
    com.intellij.openapi.application.impl.RwLockHolder.runReadAction (RwLockHolder.kt:271)
    com.intellij.openapi.application.impl.ApplicationImpl.runReadAction (ApplicationImpl.java:845)
    cursive.namespace$dependency_map.invokeStatic (namespace.clj:41)
    cursive.namespace$dependency_map.invoke (namespace.clj:35)
    cursive.repl.actions$load_file_BANG_.invokeStatic (actions.clj:397)
    cursive.repl.actions$load_file_BANG_.invoke (actions.clj:378)
    cursive.repl.actions$load_file$fn__10739.invoke (actions.clj:496)
    cursive.repl$submit$fn__1247.invoke (repl.clj:223)
    clojure.lang.AFn.run (AFn.java:22)
    com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run (RwLockHolder.kt:154)
    java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    com.intellij.util.concurrency.ContextCallable.call (ContextCallable.java:32)
    java.util.concurrent.FutureTask.run (FutureTask.java:264)
    com.intellij.util.concurrency.ContextRunnable.run (ContextRunnable.java:27)
    java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run (Executors.java:702)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run (Executors.java:699)
    java.security.AccessController.doPrivileged (AccessController.java:399)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1.run (Executors.java:699)
    java.lang.Thread.run (Thread.java:840)

java.lang.Throwable: Error evaluating
java.lang.IllegalArgumentException: Argument for @NotNull parameter 'filter' of com/intellij/util/indexing/FileBasedIndexEx.getValues must not be null
 at com.intellij.util.indexing.FileBasedIndexEx.$$$reportNull$$$0 (FileBasedIndexEx.java:-1)
    com.intellij.util.indexing.FileBasedIndexEx.getValues (FileBasedIndexEx.java:-1)
    cursive.index$namespace_names.invokeStatic (index.clj:489)
    cursive.index$namespace_names.invoke (index.clj:483)
    cursive.namespace$dependency_map$reify__4616.compute (namespace.clj:42)
    com.intellij.openapi.application.impl.RwLockHolder.runReadAction (RwLockHolder.kt:271)
    com.intellij.openapi.application.impl.ApplicationImpl.runReadAction (ApplicationImpl.java:845)
    cursive.namespace$dependency_map.invokeStatic (namespace.clj:41)
    cursive.namespace$dependency_map.invoke (namespace.clj:35)
    cursive.repl.actions$load_file_BANG_.invokeStatic (actions.clj:397)
    cursive.repl.actions$load_file_BANG_.invoke (actions.clj:378)
    cursive.repl.actions$load_file$fn__10739.invoke (actions.clj:496)
    cursive.repl$submit$fn__1247.invoke (repl.clj:223)
    clojure.lang.AFn.run (AFn.java:22)
    com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run (RwLockHolder.kt:154)
    java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:539)
    com.intellij.util.concurrency.ContextCallable.call (ContextCallable.java:32)
    java.util.concurrent.FutureTask.run (FutureTask.java:264)
    com.intellij.util.concurrency.ContextRunnable.run (ContextRunnable.java:27)
    java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1136)
    java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:635)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run (Executors.java:702)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run (Executors.java:699)
    java.security.AccessController.doPrivileged (AccessController.java:399)
    java.util.concurrent.Executors$PrivilegedThreadFactory$1.run (Executors.java:699)
    java.lang.Thread.run (Thread.java:840)

	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:376)
	at cursive.logging$do_error.invokeStatic(logging.clj:60)
	at cursive.logging$do_error.invoke(logging.clj:59)
	at cursive.logging$do_log.invokeStatic(logging.clj:45)
	at cursive.logging$do_log.doInvoke(logging.clj:38)
	at clojure.lang.RestFn.invoke(RestFn.java:467)
	at cursive.repl$print_exception.invokeStatic(repl.clj:187)
	at cursive.repl$print_exception.invoke(repl.clj:185)
	at cursive.repl$submit$fn__1247.invoke(repl.clj:225)
	at clojure.lang.AFn.run(AFn.java:22)
	at com.intellij.openapi.application.impl.RwLockHolder$executeOnPooledThread$1.run(RwLockHolder.kt:154)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
	at com.intellij.util.concurrency.ContextCallable.call(ContextCallable.java:32)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.util.concurrency.ContextRunnable.run(ContextRunnable.java:27)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
2024-07-19 08:33:19,847 [6303636] SEVERE - #cursive.repl - IntelliJ IDEA 2024.1.3  Build #IU-241.17890.1
2024-07-19 08:33:19,847 [6303636] SEVERE - #cursive.repl - JDK: 17.0.11; VM: OpenJDK 64-Bit Server VM; Vendor: JetBrains s.r.o.
2024-07-19 08:33:19,847 [6303636] SEVERE - #cursive.repl - OS: Windows 11
2024-07-19 08:33:19,847 [6303636] SEVERE - #cursive.repl - Plugin to blame: Cursive version: 1.13.4-eap7-2024.1
2024-07-19 08:33:19,847 [6303636] SEVERE - #cursive.repl - Last Action: :cursive.repl.actions/load-file
```

**Comment by cursive-ide** [comment]
I've dug through the code using the stacktrace, and I can't for the life of me see how this would happen. It looks like this is an nREPL REPL - is it JVM, or is there anything else in the mix (e.g. CLJS or Babashka)? It looks like that exception is basically immediately after the REPL starts, is that correct? i.e. you start the REPL, try to load a file, and get this exception?

**Comment by MrGung** [comment]
Yes, it is immediately, as you mention.

And yes, it's a nREPL with babashka (I mendtion bb in my first message).

**Comment by cursive-ide** [comment]
Ah, my apologies - I will see if I can repro that.

**Comment by cursive-ide** [comment]
Ok, I have managed to reproduce this. It's because the babashka library is not created correctly due to the lack of internet.

Honestly, I'm not sure how to solve this. If you want to install babashka on your machine, how would you do that? At some point you must have some process for getting things installed. Someone in a similar situation a while ago asked for a way to force-download everything that was required while their computer was connected to the internet so that they could then use it offline - would that help?

**Comment by cursive-ide** [comment]
In the meantime, I have patched the immediate source of the error for the next build.

**Comment by cursive-ide** [comment]
Fix released in 1.13.4-eap8 on 2024-07-23.

**Comment by MrGung** [comment]
> Ok, I have managed to reproduce this. It's because the babashka library is not created correctly due to the lack of internet.
> 
> Honestly, I'm not sure how to solve this. If you want to install babashka on your machine, how would you do that? At some point you must have some process for getting things installed. Someone in a similar situation a while ago asked for a way to force-download everything that was required while their computer was connected to the internet so that they could then use it offline - would that help?

I manually download babashka. IntelliJ-Marketplace is proxied - so these downloads work. I also manually download dependencies that are clojars only on another machine (don't ask... ;-))

**Comment by MrGung** [comment]
Loading a file works.


Thanks!

**Comment by cursive-ide** [comment]
> I also manually download dependencies that are clojars only on another machine (don't ask... ;-)

Ok, I won't ask :-). 

To make this work correctly, on that machine you can download the dependencies that are the result of `bb print-deps --format deps`, which should generate a deps.edn. That's what is required for all this to work properly, if you can install those in your .m2 repo, then all should work.

**Comment by cursive-ide** [comment]
Fix released in 1.13.4 on 2024-08-06.
